snap7/partner.py

```diff
@@ -100,14 +101,15 @@ def create(self, active=False):
         self._library.Par_Create.restype = snap7.types.S7Object
         self._pointer = snap7.types.S7Object(self._library.Par_Create(int(active)))
 
-    def destroy(self):
+    def destroy(self) -> Optional[c_void_p]:
         """
         Destroy a Partner of given handle.
         Before destruction the Partner is stopped, all clients disconnected and
         all shared memory blocks released.
         """
-        if self._library:
+        if self._library is not None and self._pointer is not None:
```

Comment:
Mypy had complained a lot, that self._library is initiated as None, so I may have been in autopilot and just added it this way. Should still work without the "is not Nones", since even more possibilities are checked (0, False ...).

Reply:
Okay I checked it and test_gc fails (delete a Partner object and check if just called once). This is caused by self._pointer.
If I would just use `self._library` it would complain, that byref expects a pointer (because `self._pointer` may be None), so I have to add `self._pointer `as well in the check.
Now if I would only check `if self._ponter`, it would check if Python is zero, False and more. And the pointer is pointing to None (so a NULL pointer)...which confuses me, because I would interpret it, that a real snap7 Partner (so on c-level) was never created. 😕